### PR TITLE
Fixed initialization issues with classes.

### PR DESCRIPTION
**Pull request by JessePri**
Most of the issues are fixed however some are left alone in the case of nontrivial types and classes.

### CODE REVIEW

**Comment by clayjohn** [comment]
Please see our guidelines for hacktoberfest contributors https://github.com/godotengine/godot/issues/66666

File: core/math/static_raycaster.h [hunk 1]
```diff
@@ -80,20 +80,20 @@ class StaticRaycaster : public RefCounted {
 
 	public:
 		Vector3 org; //!< Ray origin + tnear
-		float tnear; //!< Start of ray segment
+		float tnear = 0; //!< Start of ray segment
```
Comment (akien-mga) [inline core/math/static_raycaster.h:83]: Floats should be initialized with float literals `0.0f`.

You should also sync these values with what is actually being done in the constructor right above. The intializations from the constructor with constant expressions can be moved to the declarations.

**Comment by akien-mga** [comment]
> Please see our guidelines for hacktoberfest contributors #66666

I think it's ok in this case, missing initializations are bugs, unless explicitly done for performance (and then they need to be clearly labelled). But it's not necessarily as easy a task for beginners as some of our issues might make it out to be, I'll have to double check that.

**Comment by akien-mga** [comment]
The PR was abandoned.

**Pull request closed without merging**